webhook_features/src/features/summary_comment.rs

```diff
@@ -313,87 +307,29 @@ struct Review {
-}
-
 #[derive(Debug, PartialEq)]
 struct AckCommit {
     ack_type: AckType,
     commit: Option<String>,
 }
 
 fn parse_review(comment: &str) -> Option<AckCommit> {
-    let comment = comment.to_lowercase();
-    let words = comment
+    let lines = comment
         .split('\n')
-        .filter(|s| !s.starts_with('>')) // Ignore quoted text
-        .flat_map(|s| s.split(|c: char| c.is_whitespace() || c.is_ascii_punctuation())) // Split on whitespace and punctuation
-        .collect::<Vec<_>>(); // Collect into a Vec
-
-    // Split words by whitespace and punctuation
-
-    let mut pos = 0;
-    while pos < words.len() {
-        for (pattern, ack_type) in ACK_PATTERNS.iter() {
-            let pattern_words = pattern.split_whitespace().collect::<Vec<_>>(); // Split pattern into words (e.g "code review ack" => ["code", "review", "ack"])
-
-            let pattern_len = {
-                match ack_type.requires_commit_hash() {
-                    true => pattern_words.len() + 1, // If the ack type requires a commit hash, the pattern will be one word longer
-                    false => pattern_words.len(),
-                }
-            };
-            if pattern_len > words.len() - pos {
-                // If the pattern is longer than the remaining words, skip it
-                continue;
-            }
-
-            let mut matches = true;
-            for (i, pattern_word) in pattern_words.iter().enumerate() {
-                // Check if the pattern matches the words
-
-                // Ignore "re" prefixes, e.g. "reack" => "ack"
-                let mut word = words[pos + i].trim_start_matches("re-");
-                if word != "review" {
-                    word = word.trim_start_matches("re");
-                }
+        .filter(|s| !s.starts_with('>'));
 
-                if pattern_word != &word {
-                    matches = false;
-                    break;
-                }
-            }
-
-            if matches {
-                let mut commit = None;
-                if pos + pattern_words.len() < words.len() {
-                    // If there are more words after the pattern, check if the next word is a commit hash
-                    let next_word = words[pos + pattern_words.len()];
-                    if is_commit_hash(next_word) {
-                        commit = Some(next_word.to_string()); // If there is a commit hash, attach it to the ack
-                    }
-
-                    if ack_type.requires_commit_hash() && commit.is_none() {
-                        // If the ack type requires a commit hash, but there is no commit hash, skip this pattern
-                        continue;
-                    }
-                }
+    for (pattern, ack_type) in ACK_PATTERNS.iter() {
+        let re = Regex::new(pattern).unwrap();
 
+        for line in lines.clone() {
+            if let Some(caps) = re.captures(line) {
+                let commit = caps.get(2).map(|m| m.as_str().trim().to_string());
```

Comment:
```suggestion
                let commit = caps.get(2).map(|m| m.as_str().to_string());
```

(with the group change above)